### PR TITLE
Do not offer to simplify ToString on ref structs

### LINKED ISSUE #42936: IDE0071 (Interpolation can be simplified) - invalid suggestion on ref structs
**Version Used**: VS 16.5.0

**Steps to Reproduce**:

```c#
ReadOnlySpan<char> span = "test";
var str = $"{span.ToString()}"; // IDE0071 
```

**Expected Behavior**:

Do not suggest IDE0071 since removing `ToString()` is invalid for ref structs.

**Actual Behavior**:

IDE0071 is suggested.

### PR DESCRIPTION
**Pull request by jnm2**
Fixes #42936

@CyrusNajmabadi

### CODE REVIEW

File: src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs [hunk 1]
```diff
@@ -73,7 +73,8 @@ private static IOperation Unwrap(IOperation expression)
         {
             if (expression is IInvocationOperation { TargetMethod: { Name: nameof(ToString) } } invocation &&
                 HasNonImplicitInstance(invocation) &&
-                !syntaxFacts.IsBaseExpression(invocation.Instance.Syntax))
+                !syntaxFacts.IsBaseExpression(invocation.Instance.Syntax) &&
+                !invocation.Instance.Type.IsRefLikeType)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs:77]: slightly worried that invocation.Instance might be null.  but i can't htink of a way to make that happen :)

Comment (jnm2) [reply src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs:77]: The HasNonImplicitInstance would seem to guarantee that there is an instance?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs:77]: great!

**Comment by CyrusNajmabadi** [comment]
thanks!

**Pull request merged**